File: back/src/main/java/com/ssafy/recode/domain/feed/entity/Comment.java
package com.ssafy.recode.domain.feed.entity;

import com.ssafy.recode.domain.note.entity.Note;
import com.ssafy.recode.domain.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name = "comments")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long commentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "note_id")
    private Note feed;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    private String content;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

}

File: front/src/types/comment.ts
// 댓글 user interface
export interface User {
  userId: number;
  bojId: string;
  nickname: string;
  userTier: number;
}

// 댓글 type
export interface CommentResponseDTO {
  commentId: number;
  user: User;
  noteId: number;
  content: string;
  createdAt: number;
  updatedAt?: number | null;
}

export interface CommentApiResponse {
  details: CommentResponseDTO[];
}
